pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -270,620 +270,619 @@ var (
+					Schema: &EgressRule,
+				},
+			},
+			"endpointSelector": EndpointSelector,
+			"ingress": {
+				Description: "Ingress is a list of IngressRule which are enforced at ingress. " +
+					"If omitted or empty, this rule does not apply at ingress.",
+				Type: "array",
+				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+					Schema: &IngressRule,
 				},
-				"namespace": {
-					Type: "string",
+			},
+			"labels": {
+				Description: "Labels is a list of optional strings which can be used to " +
+					"re-identify the rule or to store metadata. It is possible to lookup or " +
+					"delete strings based on labels. Labels are not required to be unique, " +
+					"multiple rules can have overlapping or identical labels.",
+				Type: "array",
+				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+					Schema: &Label,
 				},
 			},
 		},
-		"spec": {
-			Description: "Spec is the desired Cilium specific rule specification.",
-			Type:        "object",
-			Ref:         getStr("#/properties/Rule"),
+	}
+
+	Service = apiextensionsv1beta1.JSONSchemaProps{
+		Description: "Service wraps around selectors for services",
+		Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+			"k8sService":         K8sServiceNamespace,
+			"k8sServiceSelector": ServiceSelector,
 		},
-		"specs": {
-			Description: "Specs is a list of desired Cilium specific rule specification.",
-			Type:        "array",
-			Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
-				Schema: &apiextensionsv1beta1.JSONSchemaProps{
-					Ref: getStr("#/properties/Rule"),
-				},
+	}
+
+	ServiceSelector = apiextensionsv1beta1.JSONSchemaProps{
+		Description: "ServiceSelector is a label selector for k8s services",
+		Required: []string{
+			"selector",
+		},
+		Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+			"selector": EndpointSelector,
+			"namespace": {
+				Type: "string",
 			},
 		},
 	}
+
+	spec = Rule
+
+	specs = apiextensionsv1beta1.JSONSchemaProps{
+		Description: "Specs is a list of desired Cilium specific rule specification.",
+		Type:        "array",
+		Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+			Schema: &Rule,
+		},
+	}
 )
+
+func init() {
+	EndpointSelector.Description = "EndpointSelector is a wrapper for k8s LabelSelector."
+
+	portRuleProps := PortRule.Properties["rules"]
+	portRuleProps.Description = "Rules is a list of additional port level rules which must be " +
+		"met in order for the PortRule to allow the traffic. If omitted or empty, " +
+		"no layer 7 rules are enforced."
+	PortRule.Properties["rules"] = portRuleProps
+
+	ruleProps := Rule.Properties["endpointSelector"]
+	ruleProps.Description = "EndpointSelector selects all endpoints which should be subject " +
+		"to this rule. Cannot be empty."
+	Rule.Properties["endpointSelector"] = ruleProps
+
+	serviceProps := Service.Properties["k8sServiceSelector"]
+	serviceProps.Description = "K8sServiceSelector selects services by k8s labels. " +
```

Comment:
@nebril wasn't you PR merged with this description changed?

Reply:
oh, was it today? I might have started with an old master

Reply:
No, it was a couple weeks ago. https://github.com/cilium/cilium/pull/2332.